ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/PlanNode.java

```diff
@@ -103,7 +103,9 @@ public DataSourceNode getLeftmostSourceNode() {
       final Optional<SourceName> sourceName
   ) {
     return getSources().stream()
-        .filter(s -> !sourceName.isPresent() || sourceName.equals(s.getSourceName()))
+        .filter(s -> !sourceName.isPresent()
+            || !s.getSourceName().isPresent()
+            || sourceName.equals(s.getSourceName()))
```

Comment:
To understand the solution, first understand the logical model of the query:

```
Source A            Source B
   |                   |
   --------Join---------
             |
       Where Filter
             |
       Final Projection
```

The final SELECT / projection node calls its parent to resolve any _select stars_.  In the case of `A.*` it passes `A` as the source name to expand when calling `resolveSelectStart`.

The parent WHERE filter uses the default impl of the `resolveSelectStar` method, which just looks for a parent node with a matching source name, if one is provided.

A Join node does not have a source name, i.e. it's neither `A` or `B`, so the default `resolveSelectStart` was ignoring it.

The fix is to change the default method to include any parent node that has no source name, thereby including the parent join node.